.github/workflows/build_wheels_macos.yml

```diff
@@ -0,0 +1,119 @@
+name: Build MacOS wheels
+
+on:
+  push:
+    branches:
+      - main # just build the sdist & wheel, skip release
+    tags:
+      - "*"
+  pull_request: # also build on PRs touching this file
+    paths:
+      - ".github/workflows/build_wheels_macos.yml"
+      - "ci/*"
+      - "MANIFEST.in"
+      - "pyproject.toml"
+      - "setup.py"
+  workflow_dispatch:
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }} ${{ matrix.platform }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["macos-latest"]
+        platform: ["x86_64"]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Cache vcpkg
+        uses: actions/cache@v1
+        id: vcpkgcache
+        with:
+          path: "/Users/runner/.cache/vcpkg/archives"
+          key: ${{ runner.os }}-${{ matrix.platform }}-vcpkg-gdal3.4.1-dynamic
```

Comment:
Note: bumping the cache key here was necessary, because GH was finding the previous cache, but vcpkg was ignoring the contents because those were for static rather than dynamic build.  Provided _this_ build works, it should then cache the dynamic version...

Reply:
Hmm, so that means that, with the current setup, we need to bump the cache manually each time that vcpkg might be installing new libraries (eg a version bump of one of the libraries) ?

Reply:
Yes, I think so.  If my hunch about the caching is correct, we'll see builds-from-source as a symptom of vcpkg updating GDAL + deps, then we'd have to update this...